opp_path_selection/src/path_selection_artist.cpp

```diff
@@ -0,0 +1,487 @@
+/*
+ * Copyright 2018 Southwest Research Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "opp_path_selection/path_selection_artist.h"
+
+#include <pcl/conversions.h>
+#include <pcl/point_cloud.h>
+#include <pcl/PolygonMesh.h>
+#include <pcl/surface/simplification_remove_unused_vertices.h>
+
+#include <eigen_conversions/eigen_msg.h>
+#include <geometry_msgs/PointStamped.h>
+#include <opp_msgs_serialization/serialize.h>
+#include <ros/package.h>
+#include <shape_msgs/Mesh.h>
+#include <tf/tf.h>
+#include <tf_conversions/tf_eigen.h>
+
+#include "opp_path_selection/path_selector.h"
+#include "opp_path_selection/path_selector_parameters.h"
+#include <opp_msgs/GetPathSelectionMesh.h>
+
+namespace opp_path_selection
+{
+const static double TF_LOOKUP_TIMEOUT = 5.0;
+const static std::string MARKER_ARRAY_TOPIC = "path_markers";
+const static std::string CLICKED_POINT_TOPIC = "clicked_point";
+const static std::string CLEAR_PATH_POINTS_SERVICE = "clear_path_selection_points";
+const static std::string COLLECT_PATH_POINTS_SERVICE = "collect_path_selection_points";
+const static std::string path_selection_config_file = ros::package::getPath("opp_path_selection") + "/config/"
+                                                                                                    "path_selection_"
+                                                                                                    "parameters.yaml";
+
+}  // namespace opp_path_selection
+
+namespace
+{
+std::vector<visualization_msgs::Marker> makeVisual(const std::string& frame_id)
+{
+  visualization_msgs::Marker points, lines;
+  points.header.frame_id = lines.header.frame_id = frame_id;
+  points.ns = lines.ns = "path_selection";
+  points.action = lines.action = visualization_msgs::Marker::ADD;
+
+  // Point specific properties
+  points.id = 0;
+  points.type = visualization_msgs::Marker::SPHERE_LIST;
+  points.scale.x = points.scale.y = 0.1;
+  points.color.b = points.color.a = 1.0;
+  points.pose.orientation.w = 1.0;
+
+  // Line specific properties
+  lines.id = 1;
+  lines.type = visualization_msgs::Marker::LINE_STRIP;
+  lines.scale.x = lines.scale.y = 0.05;
+  lines.color.b = lines.color.a = 1.0;
+  lines.pose.orientation.w = 1.0;
+
+  std::vector<visualization_msgs::Marker> visuals;
+  visuals.push_back(points);
+  visuals.push_back(lines);
+  return visuals;
+}
+
+bool pclToShapeMsg(const pcl::PolygonMesh& pcl_mesh, shape_msgs::Mesh& mesh_msg)
```

Comment:
I'm not sure if it's worthwhile to do now, but it would probably be much easier to use the `pcl_msgs/PolygonMesh` representation of a mesh instead of `shape_msgs/Mesh`